docs/blueprints/EP034.rst

```diff
@@ -0,0 +1,128 @@
+:EP: 34
+:Title: MEF E-LINE CSPF integration
+:Authors:
+    - Vinicius Arcanjo <vindasil AT fiu DOT edu>
+    - Italo Valcy <idasilva AT fiu DOT edu>
+    - Jeronimo Bezerra <jbezerra AT fiu DOT edu>
+    - Rogerio Motitsuki <rogerio.motitsuki AT gmail DOT com>
+:Created: 2022-09-26
+:Kytos-Version:
+:Status: Draft
+
+***********************************
+EP033 - MEF E-LINE CSPF integration
+***********************************
+
+Abstract
+========
+
+This blueprint proposes how ``mef_eline`` will integrate with ``pathfinder`` to request CSPF paths.
+
+Motivation
+==========
+
+Production networks typically have multiple paths in the core of the network, often times, it can be desirable to have additional constraints (the constraints and variables and ``pathfinder`` exposes) when finding a new path to account for certain business or traffic-engineering requirements.
+
+Rationale
+=========
+
+In this iteration, the major goal is to integrate ``mef_eline`` with ``pathfinder`` and its existing filtering capabilities. ``mef_eline`` relies on ``pathfinder`` to find paths for dynamic primary paths and failover paths. This proposed implementation will allow the constraints to be set individually for both primary and failover to allow it to evolve in a way where constraint requirements can be different to target a strategic set of network paths. The failover path plays a critical role in providing fast convergence, the failover path will still provide the same type of protection maximizing link disjointness but it'll compute it on top of any failover path constraints that might have been set. In the future, more granular control for how the failover path computes its disjointness might be available depending on production use cases. On ``mef_eline`` UI, users might not need to specify optional constraints both cases, we can have a strategic UX/UI, we can start with a case where if any optional constraint is given then parametrize to both dynamic primary and secondary, and gradually allowing users to have more control when ever needed.
```

Comment:
@rmotitsuki, I've added you as co-author to also help out with defining UX/UI for this integration, on `pathfinder` we currently have these components, we'll need your expertise to come up with a strategic UX/UI for this, let me know if you can help out with this, thanks:
![20220928_164850](https://user-images.githubusercontent.com/1010796/192875336-60dd4788-8421-4436-add4-e97e82f3c443.png)

Reply:
After some thought, I came up with a suggestion.
This form is complicated, with many options, and the width of the kytos menu is narrow.
We could create some kind of visual pagination.
Set the main filters on the first page and put a 'more options' link/button.
Create a quick "transition", with horizontal animation:
https://codepen.io/alvarotrigo/pen/mdpYaBB
or a vertical animation (easier):
http://jsfiddle.net/YYPKM/1677/

Reply:
@rmotitsuki, right. Sounds great to me. I liked this vertical animation that it gives this user experience that the attributes can be set gradually in steps, and then we group then strategically (for instance, in the first group with the current toolbar on mef_eline, in the second maybe only the `primary_constraints` and in the second `secondary_constraints`).

@italovalcy @jab1982 when you have a chance, let us know what you think about this initial idea since you guys will eventually use this UI in production (in addition to the APIs as well with other tools like evc_manager)

Reply:
I liked the idea of pagination!

Reply:
Thanks Italo and Rogerio. @rmotitsuki FYI, I've captured this thread on this issue https://github.com/kytos-ng/mef_eline/issues/201 and associated it with `epic_ui_mef_eline`